### PR TITLE
etcdserver: improve Lease http path naming for gRPC gateway

### LINKED ISSUE #9430: Incorrect URIs for some lease related functions in JSON API
Some Lease related URIs for the HTTP/JSON API appear to be incorrect in the generated Swagger API definitions at: https://github.com/coreos/etcd/blob/master/Documentation/dev-guide/apispec/swagger/rpc.swagger.json

**Correct URIs:** 

- "/v3/lease/grant"
- "/v3/lease/keepalive"

**Possibly incorrect URIs:** 

- "/v3/kv/lease/leases"
- "/v3/kv/lease/revoke"
- "/v3/kv/lease/timetolive"

As per the API guide, I think these should not be under the "kv" service. Not sure if the problem is just with the Swagger JSON but also in the code.

### PR DESCRIPTION
**Pull request by hexfusion**
This PR looks to normalize some of the lease api paths for the gRPC gateway. Using  additional_bindings supported by grpc-gateway we can accept both old and new paths. Current plan is to support in 3.4 and deprecate all additional_bindings in 3.5. I will work on notifying existing clients of the change as well.

fixes #9430 

/cc @akashbaid

### CODE REVIEW

**Comment by hexfusion** [comment]
I have no excuse to not add curl leases tests here ...

File: Documentation/dev-guide/apispec/swagger/rpc.swagger.json [hunk 1]
```diff
@@ -615,17 +615,7 @@
           "Lease"
         ],
         "summary": "LeaseLeases lists all existing leases.",
-        "operationId": "LeaseLeases",
-        "parameters": [
```
Comment (akashbaid) [inline Documentation/dev-guide/apispec/swagger/rpc.swagger.json]: The parameters blocks such as this getting deleted in the auto-generated file doesn't seem right. Maybe a problem in the way "additional-bindings" work?

Comment (hexfusion) [reply Documentation/dev-guide/apispec/swagger/rpc.swagger.json]: Yeah something is amiss, it works but not consistently. Possible upstream bug, but not sure.

**Comment by gyuho** [comment]
@hexfusion Can we also update changelogs?

**Comment by hexfusion** [comment]
@gyuho sure would you mind reviewing #9460 we need to address semaphore timeouts. Also I have experienced this same code above, passing and then failing Lease tests so more review is required before merge. #9460 will help this effort, thanks!

**Comment by hexfusion** [comment]
@akashbaid, I find the below to be stable and produce proper Swagger. WIll update PR and tests once https://github.com/coreos/etcd/pull/9460 gets merged.

```
  // LeaseLeases lists all existing leases.                                              
  rpc LeaseLeases(LeaseLeasesRequest) returns (LeaseLeasesResponse) {                    
      option (google.api.http) = {                                                       
        post: "/v3/lease/leases"                                                         
        body: "*"                                                                        
        additional_bindings {                                                            
            post: "/v3/kv/lease/leases"                                                  
            body: "*"                                                                    
        }                                                                                
    };                                                                                   
  }
```

**Comment by akashbaid** [comment]
That's great! Thanks for looking into it.

**Comment by hexfusion** [comment]
@gyuho updated release notes and tweaked lease tests for coverage. Should be good on greens.

**Pull request merged**